File: lldb/test/API/lang/swift/late_dylib/loader.c
#include <dlfcn.h>
#include <stdio.h>
#include <string.h>
#include <libgen.h>

int main(int argc, const char **argv) {
  char *dylib_name = strcat(dirname(argv[0]),"/dylib.dylib");
  void *dylib = dlopen(dylib_name, RTLD_NOW);
  void (*f)() = dlsym(dylib, "f");
  f();
  return 0;
}

File: lldb/test/API/lang/swift/late_swift_dylib_clangdeps/loader.swift
import Darwin

// This lookup should fail in debugger.
let fromClang = FromClang(x: 23)
print(fromClang) // break here

// Now load the dylib.
let arg0 = CommandLine.arguments[0]
let dylibName = arg0.replacing(/a\.out$/, with: "dylib.dylib")
let dylib = dlopen(dylibName, Darwin.RTLD_NOW)
let fsym = dlsym(dylib!, "f")
typealias voidTy = @convention(c) () -> ()
let f = unsafeBitCast(fsym, to: voidTy.self)
f()
